arangod/Replication2/ReplicatedLog/LogCommon.h

```diff
@@ -399,31 +398,31 @@ struct CommitFailReason {
         NonEligibleServerRequiredForQuorum const& right) noexcept
         -> bool = default;
   };
-  std::variant<NothingToCommit, QuorumSizeNotReached,
+  std::variant<SuccessfulQuorum, QuorumSizeNotReached,
                ForcedParticipantNotInQuorum, NonEligibleServerRequiredForQuorum>
       value;
 
-  static auto withNothingToCommit() noexcept -> CommitFailReason;
+  static auto withSuccessfulQuorum() noexcept -> CommitDetails;
   static auto withQuorumSizeNotReached(ParticipantId who) noexcept
-      -> CommitFailReason;
+      -> CommitDetails;
   static auto withForcedParticipantNotInQuorum(ParticipantId who) noexcept
-      -> CommitFailReason;
+      -> CommitDetails;
   static auto withNonEligibleServerRequiredForQuorum(
       NonEligibleServerRequiredForQuorum::CandidateMap) noexcept
-      -> CommitFailReason;
+      -> CommitDetails;
 
-  static auto fromVelocyPack(velocypack::Slice) -> CommitFailReason;
+  static auto fromVelocyPack(velocypack::Slice) -> CommitDetails;
   void toVelocyPack(velocypack::Builder& builder) const;
 
-  friend auto operator==(CommitFailReason const& left,
-                         CommitFailReason const& right) -> bool = default;
+  friend auto operator==(CommitDetails const& left, CommitDetails const& right)
+      -> bool = default;
 
  private:
   template<typename... Args>
-  explicit CommitFailReason(std::in_place_t, Args&&... args) noexcept;
+  explicit CommitDetails(std::in_place_t, Args&&... args) noexcept;
 };
 
-auto to_string(CommitFailReason const&) -> std::string;
+auto to_string(CommitDetails const&) -> std::string;
```

Comment:
The name `CommitDetails` implicates that this contains information about the last commit that happened. This confusion additionally manifests itself in renaming `NothingToCommit` to `SuccessfulQuorum` which is a statement about the last commit operation. 

The purpose of this data structure was always to explain to the customer why the system **currently does not commit** more entries. i.e. it is the answer to the question why the commit index is not higher than it currently is. So if there are no log entries pending for commit, it should say in a way, that there is nothing to commit. On the other hand, declaring that the commit index is not higher, because there was a `SuccessfulQuorum` is ... wrong?

You might as well just set it always to `SuccessfulQuorum` without lying.

I have no idea how to call that thing to prevent this confusion again.